docs/requirements/channel-upgradability-requirements.md

```diff
@@ -0,0 +1,224 @@
+- The timeout (either as block height or timestamp) until when upgrades can be initiated after approval. This can also be specified per specific pairs of port ID, channel ID (e.g. for channel (`transfer`, `channel-0`)) or for all channels using a certain port ID (e.g. for all channels on port ID `transfer`).
+
+Some concrete examples to illustrate the above:
+
+- Approve upgrading channel (`transfer`, `channel-0`) to version `ics20-2` until block height 1000.
+- Approve upgrading channels (`transfer`, `channel-0`) & (`transfer`, `channel-1`) to version `ics20-2` until block height 1000 AND channel (`icacontroller-cosmos13fdx64cc9afrsnjdytk6drnj3mzdku5qsemlz5`, `channel-2`) to ordering `UNORDERED` until time 1682288559.
+- Approve upgrading all channels on port ID `transfer` to version `{"fee_version":"ics29-1","app_version":"ics20-1"}` until block height 1050.
+- Approve upgrading all channels with port ID starting with `icacontroller` to version `{"fee_version":"ics29-1","app_version":"ics27-1"}` until block height 1080.
+
+Once an approved upgrade succeeds, it is required that a new governance or group proposal passes to approve a new upgrade to the channel.
+
+Assuming that we have two chains, and that both chains approve a compatible upgrade (i.e. the proposed upgrade parameters are identical on both chains and a channel between them is in the upgrade scope), then the following upgrade initiation combinations are possible:
+
+- Permissionless upgrade initiation on both chain A and B.
+     In this scenario both chains approve a channel upgrade and a relayer can submit `MsgChannelUpgradeInit` on one side to start the handshake witin the timeout period specified in the upgrade approval proposal. Upon successful execution of `MsgChannelUPgradeInit` a relayer can submit `MsgChannelUpgradeTry` on the counterparty.
+- Permissionless upgrade on chain A and permissioned upgrade on chain B.
+- Permissioned upgrade on chain A and permissionless upgrade on B.
+    In these 2 scenarios, both side approve a channel upgrade, but one chain automatically initiates the upgrade process by executing `MsgChannelUpgradeInit` and the other side needs a relayer to submit either `MsgChannelUpgradeInit` or `MsgChannelUpgradeTry`. If `MsgChannelUpgradeInit` is executed on both chains, then we would have a crossing hello uld be that a crossing hello situation occurs if relayer(s) submit `MsgChannelUpgradeInit` on both chains.
```

Comment:
```suggestion
    In these 2 scenarios, both sides approve a channel upgrade, but one chain automatically initiates the upgrade process by executing `MsgChannelUpgradeInit` and the other side needs a relayer to submit either `MsgChannelUpgradeInit` or `MsgChannelUpgradeTry`. If `MsgChannelUpgradeInit` is executed on both chains, then we would have a crossing hello uld be that a crossing hello situation occurs if relayer(s) submit `MsgChannelUpgradeInit` on both chains.
```

in addition, I think this sentence needs some clarity?

> then we would have a crossing hello uld be that a crossing hello situation occurs if relayer(s) submit `MsgChannelUpgradeInit` on both chains.

Reply:
Yes :) That sentence was a complete mess :D
Thanks for catching all the grammar mistakes!